### PR TITLE
Support options after RUN

### PR DESCRIPTION
**Pull request by mjambon**
The [RUN](https://docs.docker.com/engine/reference/builder/#run) instruction now supports options. This PR adds support for any future option, in addition to those officially supported at this time (`--mount`, `--network`, `--security`).

The value of the `--mount` option itself has a structure that we want to inspect [with Semgrep](https://github.com/returntocorp/semgrep/issues/6353). That's why I added a special treatment for it.

### CODE REVIEW

**Comment by mjambon** [comment]
Yesterday's commit was bogus. It's fixed now.

**Comment by mjambon** [comment]
:wave: we'd like to merge this.

File: grammar.js [hunk 1]
```diff
@@ -301,7 +333,24 @@ module.exports = grammar({
...
+        // constructs because the lexer prefers the longer tokens
+        // when it has a choice. The example below shows the tokenization
+        // of the --mount parameter.
+        //
+        //   RUN --mount=foo=bar,baz=42 ls --all
+        //       ^^     ^   ^   ^   ^
+        //         ^^^^^ ^^^ ^^^ ^^^ ^^
+        //       |--------param-------|
+        //                              |--shell_command--|
+        //
```
Comment (camdencheek) [inline grammar.js:348]: Thanks for the detailed comment

**Review by camdencheek — APPROVED** [review]
Looks good to me -- thanks for the contribution! Sorry about the delay and thanks for the ping.

**Pull request merged**